src/pg_cron.c

```diff
@@ -1267,5 +1561,536 @@ ManageCronTask(CronTask *task, TimestampTz currentTime)
 			InitializeCronTask(task, jobId);
 		}
 
+	/* keep compiler quiet */
+	(void) result;
 	}
 }
+
+static void
+GetTaskFeedback(shm_mq_handle *responseq, PGresult *result, CronTask *task)
+{
+
+	TimestampTz end_time;
+	bool update_result;
+
+	end_time = GetCurrentTimestamp();
+
+	if (responseq) {
+		Size            nbytes;
+		void       *data;
+		char            msgtype;
+		StringInfoData  msg;
+
+		/*
+		 * Message-parsing routines operate on a null-terminated StringInfo,
+		 * so we must construct one.
+		 */
+		shm_mq_receive(responseq, &nbytes, &data, false);
+		initStringInfo(&msg);
+		resetStringInfo(&msg);
+		enlargeStringInfo(&msg, nbytes);
+		msg.len = nbytes;
+		memcpy(msg.data, data, nbytes);
+		msg.data[nbytes] = '\0';
+		msgtype = pq_getmsgbyte(&msg);
+		switch (msgtype)
+		{
+			case 'E':
+				{
+					ErrorData	edata;
+					pq_parse_errornotice(&msg, &edata);
+
+					if (CronLogRun)
+					{
+						update_result = InsertOrUpdateJobRunDetail(task->runId, NULL, NULL, NULL,
+												NULL, NULL, "failed", edata.message,
+												NULL, &end_time);
+						Assert(update_result);
+					}
+
+					ereport(LOG, (errmsg("cron job " INT64_FORMAT " ERROR: %s",
+									 task->jobId, edata.message)));
+					break;
+				}
+			case 'N':
+				{
+					ErrorData	edata;
+					pq_parse_errornotice(&msg, &edata);
+
+					if (CronLogRun)
+					{
+						update_result = InsertOrUpdateJobRunDetail(task->runId, NULL, NULL, NULL,
+												NULL, NULL, "failed", edata.message,
```

Comment:
not sure this is the right thing to do.

If I do the following:
```sql
CREATE OR REPLACE FUNCTION public.test()
RETURNS void
LANGUAGE plpgsql 
AS $$
DECLARE
BEGIN
        RAISE WARNING 'hello world';
END;
$$;
select cron.schedule('* * * * *', 'SELECT test()');
```

The run shows up as "failed" even though it succeeded: 
```sql
postgres=# table cron.job_run_details;
┌───────┬───────┬─────────┬──────────┬──────────┬────────────────────┬───────────┬────────────────┬───────────────────────────────┬───────────────────────────────┐
│ jobid │ runid │ job_pid │ database │ username │      command       │  status   │ return_message │          start_time           │           end_time            │
├───────┼───────┼─────────┼──────────┼──────────┼────────────────────┼───────────┼────────────────┼───────────────────────────────┼───────────────────────────────┤
│   162 │  2719 │    1197 │ postgres │ marco    │ SELECT test()      │ failed    │ hello world    │ 2020-09-01 14:53:00.005055+02 │ 2020-09-01 14:53:00.010426+02 │
└───────┴───────┴─────────┴──────────┴──────────┴────────────────────┴───────────┴────────────────┴───────────────────────────────┴───────────────────────────────┘
(1 row)
```